tests/vm_test.go

```diff
@@ -971,309 +969,149 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 		})
 	})
 
-	Context("[rfe_id:273]with oc/kubectl", func() {
-		var k8sClient string
-		var workDir string
-		var vmRunningRe *regexp.Regexp
-
-		BeforeEach(func() {
-			k8sClient = clientcmd.GetK8sCmdClient()
-			clientcmd.SkipIfNoCmd(k8sClient)
-			workDir = GinkgoT().TempDir()
-
-			// By default "." does not match newline: "Phase" and "Running" only match if on same line.
-			vmRunningRe = regexp.MustCompile("Phase.*Running")
-		})
-
-		createVMAndGenerateJson := func(opts ...libvmi.VMOption) (*v1.VirtualMachine, string) {
-			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine(), opts...)
-			vm.Namespace = testsuite.GetTestNamespace(vm)
-
-			vmJson, err := tests.GenerateVMJson(vm, workDir)
-			Expect(err).ToNot(HaveOccurred(), "Cannot generate VMs manifest")
-
-			return vm, vmJson
-		}
-
-		It("[test_id:243][posneg:negative]should create VM only once", decorators.Conformance, func() {
-			vm, vmJson := createVMAndGenerateJson(libvmi.WithRunStrategy(v1.RunStrategyAlways))
-
-			By("Creating VM with DataVolumeTemplate entry with k8s client binary")
-			_, _, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
+	Context("should not change anything if dry-run option is passed", func() {
+		It("[test_id:7530] when starting a VM", func() {
+			By("Creating VM")
+			vm := libvmi.NewVirtualMachine(libvmifact.NewAlpine())
+			vm, err := virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm, metav1.CreateOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			By("Verifying VM is created")
-			vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred(), "New VM was not created")
-
-			By("Creating the VM again")
-			_, stdErr, err := clientcmd.RunCommand(testsuite.GetTestNamespace(nil), k8sClient, "create", "-f", vmJson)
-			Expect(err).To(HaveOccurred())
+			By("Performing dry run start")
+			err = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{DryRun: []string{metav1.DryRunAll}})
```

Comment:
Two things:

1. This is a potential race condition and probably should use Consistently
2. You could also assert on vm.Status.Created additionally/instead

Reply:
I think the Start/Stop/Restart DryRUn test cases should be moved to `tests/subresource_api_test.go`. They are verifying the subresource API works as intended.